### PR TITLE
bpo-45155: Apply new byteorder default values for int.to/from_bytes

### PR DESCRIPTION
**Pull request by rhettinger**
The code is a little faster and a little more readable with the new default value for byteorder.

Added comments to point of endianess in cases where it might matter.

https://bugs.python.org/issue45155

### CODE REVIEW

File: Lib/uuid.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def __init__(self, hex=None, bytes=None, bytes_le=None, fields=None,
             if len(bytes) != 16:
                 raise ValueError('bytes is not a 16-char string')
             assert isinstance(bytes, bytes_), repr(bytes)
-            int = int_.from_bytes(bytes, byteorder='big')
+            int = int_.from_bytes(bytes)  # big_endian
```
Comment (vstinner) [inline Lib/uuid.py]:
```suggestion
            int = int_.from_bytes(bytes)  # big endian
```

**Review by vstinner — APPROVED** [review]
LGTM, I just added a minor suggestion.

For the commit message, you can reuse "bpo-45155: " to give more context to the change.

File: Lib/uuid.py (hunk 1 above)
Comment (ambv) [reply Lib/uuid.py]: `big_endian` can be improved, yes. But it seems to me the adjective form is spelled with a hyphen (`big-endian`). CPython's got 97 examples of that spelling already versus only 33 "[Bb]ig endian" with space.

```suggestion
            int = int_.from_bytes(bytes)  # big-endian
```

File: Lib/hashlib.py [hunk 2]
```diff
@@ -235,15 +235,15 @@ def prf(msg, inner=inner, outer=outer):
         loop = 1
         from_bytes = int.from_bytes
         while len(dkey) < dklen:
-            prev = prf(salt + loop.to_bytes(4, 'big'))
+            prev = prf(salt + loop.to_bytes(4))  # big endian
             # endianness doesn't matter here as long to / from use the same
-            rkey = int.from_bytes(prev, 'big')
+            rkey = int.from_bytes(prev)
```
Comment (serhiy-storchaka) [inline Lib/hashlib.py]:
```suggestion
            rkey = from_bytes(prev)
```

File: Lib/hashlib.py [hunk 3]
```diff
@@ -235,15 +235,15 @@ def prf(msg, inner=inner, outer=outer):
         loop = 1
         from_bytes = int.from_bytes
         while len(dkey) < dklen:
-            prev = prf(salt + loop.to_bytes(4, 'big'))
+            prev = prf(salt + loop.to_bytes(4))  # big endian
```
Comment (serhiy-storchaka) [inline Lib/hashlib.py]: The comment is not needed here. Se the one at the next line.
```suggestion
            prev = prf(salt + loop.to_bytes(4))
```

File: Lib/uuid.py (hunk 1 above)
Comment (ambv) [reply Lib/uuid.py]: In fact, wait... What's the point of applying the default if you have to add a comment saying what that default is? In this case either we should trust the default to be obvious, or just not use it and be explicit instead, right?

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**